gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ConditionMapper.java

```diff
@@ -0,0 +1,242 @@
+        Reference reference = buildResourceReference(actualProblem);
+        return buildReferenceExtension(ACTUAL_PROBLEM_URL, reference);
+    }
+
+    private Extension buildProblemSignificance(CD linksetCode) {
+        Optional<String> code = Optional.ofNullable(linksetCode.getQualifier().get(0).getName().getCode());
+        Extension extension = new Extension();
+        extension.setUrl(PROBLEM_SIGNIFICANCE_URL);
+
+        if (code.isPresent() && code.get().equals(MAJOR_CODE)) {
+            extension.setValue(new CodeType("Major"));
+        } else {
+            extension.setValue(new CodeType("Minor"));
+        }
+
+        return extension;
+    }
+
+    private Optional<Condition.ConditionClinicalStatus> buildClinicalStatus(CD linksetCode) {
+        if (CLINICAL_STATUS_ACTIVE_CODE.equals(linksetCode.getCode())) {
+            return Optional.of(Condition.ConditionClinicalStatus.ACTIVE);
+        } else if (CLINICAL_STATUS_INACTIVE_CODE.equals(linksetCode.getCode())) {
+            return Optional.of(Condition.ConditionClinicalStatus.INACTIVE);
+        }
+        return Optional.empty();
+    }
+
+    private CodeableConcept generateCategory() {
+        Coding coding = new Coding();
+        coding.setSystem("https://fhir.hl7.org.uk/STU3/CodeSystem/CareConnect-ConditionCategory-1");
+        coding.setCode("problem-list-item");
+        coding.setDisplay("Problem List Item");
+
+        return new CodeableConcept()
+            .addCoding(coding);
+    }
+
+    private List<Extension> buildRelatedClinicalContent(List<Resource> relatedClinicalResources) {
+        return relatedClinicalResources.stream()
+            .map(ResourceUtil::buildResourceReference)
+            .map(reference -> buildReferenceExtension(RELATED_CLINICAL_CONTENT_URL, reference))
+            .collect(Collectors.toList());
+    }
+
+    private List<Annotation> buildNotes(Optional<RCMRMT030101UK04ObservationStatement> observationStatement,
+        RCMRMT030101UK04LinkSet linkSet) {
+        List<Annotation> annotationList = new ArrayList<>();
+
+        observationStatement.ifPresent(observationStatement1 -> observationStatement1.getPertinentInformation()
+            .stream()
+            .map(RCMRMT030101UK04PertinentInformation02::getPertinentAnnotation)
+            .filter(Objects::nonNull)
+            .map(RCMRMT030101UK04Annotation::getText)
+            .filter(StringUtils::isNotBlank)
+            .map(StringType::new)
+            .map(Annotation::new)
+            .forEach(annotationList::add));
+
+        if (linkSet.getCode().getOriginalText() != null) {
+            StringType text = new StringType(linkSet.getCode().getOriginalText());
+            annotationList.add(new Annotation(text));
+        }
+
+        return annotationList;
+    }
+
+    @Builder
+    @Getter
+    public static class ConditionMapperParameters {
+        private RCMRMT030101UK04EhrComposition ehrComposition;
```

Comment:
I think that it would be easier to pass entire FHIR as params. E.g You could pass Patient resource, not patientId. It would be ConditionMapper's responsibility to parse it end extract what it needs. Also - you wouldn't need ResourceUtil.buildResourceReference. You could just new new Reference(patientResource).

In general - I think that we should make the mapper as easy to use as possible. If it needs multiple XML elements - pass entire EhrExtract, if it needs single values from other, already mapped resources - pass those resources and encapsulate the logic to extract those values in ConditionMapper.

I'm happy to discuss it if needed.